### ISSUE TITLE
Incorrect suggestion to replace hPutStrLn with putStrLn

### ISSUE THREAD

**Issue by ocharles** [ndmitchell/hlint#294]
```haskell
{-# LANGUAGE NoImplicitPrelude #-}
module Test where

import Data.IOData
import System.IO (stdout)

main :: IO ()
main = hPutStrLn stdout "Hello, World!"
```

This `hPutStrLn` comes from `Data.IOData` (in `chunked-data`), which has different behavior to the suggested `putStrLn` (which isn't even in scope, due to `NoImplicitPrelude`).

**Comment by ndmitchell** [comment]
Thanks. I guess scopes based on imports, but don't currently look at the implicit prelude flag - that's the bug.
